Description: Mock openstack_dashboard secret key in unit tests
 This will write to the filesystem wherever horizon is installed, which
 isn't friendly to packagers. Make it just a boring string instead.
Author: Jim Rollenhagen <jim@jimrollenhagen.com>
Date: Wed, 28 Sep 2016 15:58:51 +0000 (-0400)
X-Git-Url: https://review.openstack.org/gitweb?p=openstack%2Fironic-ui.git;a=commitdiff_plain;h=eff21be9c605dac069332788711a63954447e3fa
Change-Id: I97fc397562487bc3a5f88fb3ac5f2b5efb41f7bb
Origin: upstream, https://review.openstack.org/378744
Last-Update: 2016-09-28

diff --git a/ironic_ui/test/settings.py b/ironic_ui/test/settings.py
index 4feed0e..6974c7a 100644
--- a/ironic_ui/test/settings.py
+++ b/ironic_ui/test/settings.py
@@ -25,7 +25,14 @@ HORIZON_CONFIG.pop('default_dashboard', None)
 # Update the dashboards with ironic_ui
 import ironic_ui.enabled
 import openstack_dashboard.enabled
-from openstack_dashboard.utils import settings
+
+# NOTE(jroll) importing this will cause it to write a key for django, if one
+# does not exist, which isn't super friendly. Hack it for testing.
+import mock
+with mock.patch(
+        'horizon.utils.secret_key.generate_or_read_from_file') as mock_key:
+    mock_key.return_value = 'test'
+    from openstack_dashboard.utils import settings
 
 settings.update_dashboards(
     [
